graylog2-web-interface/src/components/configurations/UrlWhiteListForm.tsx

```diff
@@ -211,9 +199,29 @@ const UrlWhiteListForm = ({ urls, onUpdate, disabled }: Props) => {
   };
 
   useEffect(() => {
-    const valid = _isFormValid();
+    const isNewEntryValid = async () => {
+      const newEntryIdx = config.entries.findIndex((entry) => entry.id === newEntryId);
+
+      if (newEntryIdx < 0) {
+        return false;
+      }
+
+      const newEntry = config.entries[newEntryIdx];
+      const entryValidation = await validateUrlEntry(newEntryIdx, newEntry, _updateValidationError);
+
+      return map(entryValidation, 'valid').some((valid) => !!valid);
+    };
 
-    onUpdate(config, valid);
+    const propagateUpdate = async (firstRender) => {
+      const valid = firstRender && newEntryId ? await isNewEntryValid() : hasValidationErrors();
+      onUpdate(config, valid);
+    };
+
+    propagateUpdate(!isMounted.current);
+
+    if (!isMounted.current) {
+      isMounted.current = true;
+    }
   }, [config]);
```

Comment:
The linter failure here was left on purpose. Changing the dependency array had some negative consequences in the past, as well as increasing the number of calls to `onUpdate()` drastically, so I think changing it right now may do more harm than good before the release.